sys/console/full/src/uart_console.c

```diff
@@ -252,11 +249,20 @@ uart_console_rx_char_event(struct os_event *ev)
 
         ret = console_handle_char(b);
         if (ret < 0) {
-            return;
+            goto done;
         }
     }
 
     b = -1;
+
+done:
+    /*
+     * It is possible that UART RX was stalled in the meantime since we were
+     * not consuming data from ring buffer - make sure RX is started.
+     */
+    if (full) {
```

Comment:
Perhaps to make it a bit smarter we can remember that RX was stalled in `uart_console_rx_char` (line 211) and only restart in such case. I'm not 100% sure if this won't create some race though...

Also, restart can be done after leaving critical section above (line 249) since we already have free space in ring buffer and can accept new data immediately.